logic/atrium-logic-jvm/src/main/kotlin/ch/tutteli/atrium/logic/impl/DefaultPathAssertions.kt

```diff
@@ -180,4 +183,12 @@ class DefaultPathAssertions : PathAssertions {
         other: String
     ): FeatureExtractorBuilder.ExecutionStep<T, Path> = container.f1<T, String, Path>(Path::resolve, other)
 
+    override fun <T : Path> hasDirectoryEntry(container: AssertionContainer<T>, entries: List<String>): Assertion =
+        assertionBuilder.invisibleGroup.withAssertions(
+            listOf(isDirectory(container)) +
+                entries.map { entry ->
+                    resolve(container, entry)
+                        .collect { _logic.addAssertion(exists(_logic, NOFOLLOW_LINKS)) }
+                }
```

Comment:
don't use the methods of this class directly as one might use a different implementation. Always take the indirection of _logic
```suggestion
            listOf(container.isDirectory()) +
                entries.map { entry ->
                    container.resolve(entry).collect { _logic.exists(NOFOLLOW_LINKS) }
                }
```